tests_new/integration_tests/aws_clients/iam.py

```diff
@@ -34,7 +43,9 @@ def create_role(self, account_id, role_name):
                 {{
                     "Effect": "Allow",
                     "Principal": {{
-                        "AWS": "arn:aws:iam::{account_id}:root"
+                        "AWS": ["arn:aws:iam::{account_id}:root",
+                        "arn:aws:iam::{IAMClient.get_tooling_account_id()}:root",
+                        "arn:aws:sts::{account_id}:assumed-role/{test_role_name}/{test_role_name}"]
```

Comment:
I am no IAM expert but do we need this? I think the first principal (line 46) will allow all roles from `account_id` to assume this role. Check [this](https://niebardzo.github.io/2021-08-23-root-principal-in-aws/#:~:text=IAM%20role%20assigned.-,Root%20Principal,-Who%20is%20the)

Reply:
No, it doesn't work. Assumed role is processed differently. I tried without it and got Access Denied, so I had to explicitly add this.